### PR TITLE
Deploy zkevm contracts to hotshot deployment

### PR DESCRIPTION
**Pull request by sveitser**
- Make hotshot contract deployment optional.
- Make deployment of second zkevm rollup optional.
- Log a bit more info about deployment at the start.
- Allow configuring polling interval.

Generates something like this:
```
$ cargo run --bin deploy -- --hotshot-address 0x73511669fd4de447fed18bb79bafeac93ab7f31f
# Deployment configuration (generated by deploy.rs)
ESPRESSO_SEQUENCER_HOTSHOT_ADDRESS="0x73511669fd4de447fed18bb79bafeac93ab7f31f"
ESPRESSO_ZKEVM_1_BRIDGE_ADDRESS="0x04c89607413713ec9775e14b954286519d836fef"
ESPRESSO_ZKEVM_1_CHAIN_ID=1001
ESPRESSO_ZKEVM_1_FORK_ID=1
ESPRESSO_ZKEVM_1_GENESIS_BLOCK_NUMBER=70
ESPRESSO_ZKEVM_1_GENESIS_ROOT="5c8df6a4b7748c1308a60c5380a2ff77deb5cfee3bf4fba76eef189d651d4558"
ESPRESSO_ZKEVM_1_GER_ADDRESS="0xdbc43ba45381e02825b14322cddd15ec4b3164e6"
ESPRESSO_ZKEVM_1_HOTSHOT_ADDRESS="0x73511669fd4de447fed18bb79bafeac93ab7f31f"
ESPRESSO_ZKEVM_1_MATIC_ADDRESS="0x1fa02b2d6a771842690194cf62d91bdd92bfe28d"
ESPRESSO_ZKEVM_1_NETWORK_NAME="zkevm-one"
ESPRESSO_ZKEVM_1_ROLLUP_ADDRESS="0x4c4a2f8c81640e47606d3fd77b353e87ba015584"
ESPRESSO_ZKEVM_1_TRUSTED_AGGREGATOR="0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266"
ESPRESSO_ZKEVM_1_VERIFIER_ADDRESS="0x5081a39b8a5f0e35a8d959395a630b68b74dd30f"
```

### CODE REVIEW

File: zkevm-contract-bindings/src/bin/deploy.rs [hunk 1]
```diff
@@ -10,15 +10,17 @@ use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
...
 use contract_bindings::HotShot;
 use ethers::{
+    prelude::{NonceManagerMiddleware, SignerMiddleware},
     providers::{Http, Middleware, Provider},
+    signers::{coins_bip39::English, MnemonicBuilder, Signer},
     types::Address,
     utils::{get_contract_address, parse_ether},
 };
 use hex::{FromHex, FromHexError};
-use sequencer_utils::{connect_rpc, Middleware as EthMiddleware};
```
Comment (jbearer) [inline zkevm-contract-bindings/src/bin/deploy.rs:18]: What was wrong with the `connect_rpc` from `sequencer_utils`?

**Review by jbearer — APPROVED** [review]
LGTM

File: zkevm-contract-bindings/src/bin/deploy.rs (hunk 1 above)
Comment (sveitser) [reply zkevm-contract-bindings/src/bin/deploy.rs:18]: I just couldn't figure out how to change the polling interval after that function returns the client.

**Comment by Ancient123** [comment]
Two things 
Can we have an option for it to output in `json` format?
How do I target the correct L1 ?

**Comment by sveitser** [comment]
@Ancient123 

> Two things Can we have an option for it to output in `json` format? 

Sure I can re-add that.

> How do I target the correct L1 ?

Set `ESPRESSO_ZKEVM_DEPLOY_WEB3_PROVIDER_URL` or `--provider-url` to the desired RPC URL.

**Comment by sveitser** [comment]
Added a `--json` flag. So you can do for example

```
cargo run --bin deploy -- --hotshot-address 0x73511669fd4de447fed18bb79bafeac93ab7f31f --json --output-path test.json
```

**Comment by jbearer** [comment]
This CI failure is fine, it works for me locally but often flakes in CI because the runners are sometimes slow and the test is based on timing. I would say it's ok to merge

**Comment by Ancient123** [comment]
> Added a `--json` flag. So you can do for example
> 
> ```
> cargo run --bin deploy -- --hotshot-address 0x73511669fd4de447fed18bb79bafeac93ab7f31f --json --output-path test.json
> ```

Thank you!

**Review by Ancient123 — APPROVED** [review]

**Pull request merged**